### PR TITLE
Scroll service categories to top on selection

### LINKED ISSUE #267: Tour callout offscreen when no SaaS offerings
screen shot 2017-06-02 at 2 57 23 pm

### PR DESCRIPTION
**Pull request by jeff-phillips-18**
This PR scrolls the ‘Browse Catalog’ section to the top of the landing page when a category is selected. This makes it so the page doesn't 'jump' while browsing through the available categories.

Fixes #267 

Jumping issue:

Before:

scroll-issue

After"

scroll-issue-fixed

### CODE REVIEW

**Comment by jeff-phillips-18** [comment]
Also, after further UX feedback, added the margins around the All and Other services areas to be more in line with the sub-categories areas.

**Comment by rhamilto** [comment]
> Also, after further UX feedback, added the margins around the All and Other services areas to be more in line with the sub-categories areas (not shown in the fixed image above).

screen shot 2017-06-13 at 9 18 34 am

File: src/styles/services-view.less [hunk 1]
```diff
@@ -134,6 +134,17 @@ services-view,
       }
     }
 
+    .services-no-sub-categories {
```
Comment (rhamilto) [inline src/styles/services-view.less]: Nit:  this rule should come before .services-sub-categories per alpha rule

Comment (jeff-phillips-18) [inline src/styles/services-view.less]: Fixed.

**Comment by jeff-phillips-18** [comment]
Fixed unit tests.

**Comment by beanh66** [comment]
Thanks  @jeff-phillips-18! Much improved interaction :)

**Comment by cshinn** [comment]
@jeff-phillips-18 I think it works great, but I wonder if the services area should have side and bottom padding on mobile. Seems a bit narrow.

**Comment by jwforres** [comment]
its a visual nit but it bothers me that the margin above and below items in the white section is not the same

whitespace

**Comment by spadgett** [comment]
I might have found a bug: when I click on a tab in the second roll, it scrolls the items offscreen for me:

screen shot 2017-06-13 at 9 24 56 pm

**Comment by spadgett** [comment]
I'm having trouble consistently reproducing though. It might have been because I resized the window?

**Comment by serenamarie125** [comment]
This stuff looks great!  @jeff-phillips-18 thanks for the improvements, @rhamilto definitely like the additional margins!

Autoscrolling doesn't bother me, it seems useful imo.

**Comment by jwforres** [comment]
Hmm... yeah I'm seeing some weird jumping behavior during browser resize sometimes.

**Comment by jwforres** [comment]
Ok got a screen capture of what I'm seeing, watch the bottom of the SaaS section jump around as you resize.  I suspect it has to do with how resize is being debounced.

**Comment by jeff-phillips-18** [comment]
@jwforres re: margins  Now that the filter stuff is in as well can you re-look and see if it still bothers you.

**Comment by jeff-phillips-18** [comment]
@spadgett Is this that issue w/ scrolling just due to not showing scrollbars until you hover? I notice there are no scrollbars for the project summary panel either.

**Comment by jeff-phillips-18** [comment]
@jwforres I can cut down the debounce timer maxWait, it helps a bit but there will still be some amount of jumping. Any other thoughts?

**Comment by jwforres** [comment]
Still looks bad to me.  I dont think the problem is actually new on the subcategories, it just didnt exist on the All/Other pages before because the white background went to the bottom.  The spacing between the logo and the text also seems quite large to me:

whitespace2

**Comment by jeff-phillips-18** [comment]
Rebased. Set the maxWait for the resize debounce to 50ms.

**Comment by jeff-phillips-18** [comment]
@jwforres it is a bit better with the filter layout refactoring:

image

**Comment by rhamilto** [comment]
I agree with @jwforres regarding the margin around and lack of full height for "All" and "Other".  The lack of connection to the tabs creates a disassociation between the tabs and the content (that is, they no longer feel associated).  And the white not extending all the way to the bottom of the page actually makes the items appear less important because my eye is drawn by the negative space surrounding them.

**Comment by jeff-phillips-18** [comment]
Is this something new to this PR? The only thing changed in this PR is to add the left/right margins. The top/bottom distances ought not have changed. Is there a difference in this disassociation here than there is for the sub-category tiles? It looks the same to me (maybe worse for the sub-categories since there is even more 'negative space').

**Comment by spadgett** [comment]
@jeff-phillips-18 Didn't the white extend to the bottom of the page before? I know it's an intentional change, but I think it's what causes the feeling on unbalanced margins.

For what it's worth, I preferred the All and Other categories as they were before, but I see there's a pretty strong split on that.

**Comment by jeff-phillips-18** [comment]
@spadgett yes, that too. I can go either way. I was just trying to make it consistent when clicking through the categories.

**Comment by jeff-phillips-18** [comment]
@jwforres @rhamilto @spadgett Is this more favorable?

image

**Comment by jwforres** [comment]
I definitely prefer that over the shortened white block. One nit, I'd expect the bottom gutter to be the same as the sides

**Comment by spadgett** [comment]
I prefer that, but I still like the current style better :) I think @rhamilto hit on it when he said the tabs don't feel connected to the content.

**Comment by spadgett** [comment]
I think the autoscroll is an improvement, though

**Comment by jeff-phillips-18** [comment]
I've pushed this change, with the bottom margin fix:

image

**Comment by jwforres** [comment]
i dunno how to make the resize less jumpy, so if we think the improvement we get from the auto-scroll justifies the resize behavior then I'm ok with this.  I would lean towards yes and sounds like @spadgett does?

**Comment by spadgett** [comment]
I wouldn't be against removing debounce if it's causing problems.

If we keep it, it feels more like a `_.throttle` than `_.debounce` when max wait is the same as wait.

https://lodash.com/docs/3.10.1#throttle

We could try `_.throttle` with an even smaller value or remove it altogether.

**Comment by jeff-phillips-18** [comment]
Found a way to not update on resize (using 100vh - offset)! No more jumping!

@jwforres Give it another go and see what you think

**Comment by jwforres** [comment]
soooo much better 👍

**Pull request merged**